realm/realm-library/src/androidTest/java/io/realm/RealmTest.java

```diff
@@ -2696,4 +2696,54 @@ public void execute(Realm realm) {
             handlerThread.quit();
         }
     }
+
+    public void testHandlerControllerThrowNullPointExceptionOnNonLooperThread() {
+        final CountDownLatch signalTestFinished = new CountDownLatch(1);
+        Thread thread = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                Realm realm = Realm.getInstance(testConfig);
+                try {
+                    realm.handlerController.removeAllChangeListeners();
+                    fail("");
+                } catch (NullPointerException ignored) {
```

Comment:
I think this UT is already covered by the test below. The path we need to test is calling `removeAllChangeListeners` on a non-looper thread should throw. Maybe refactor this test to cover  call to `removeChangeListener` instead?